ng-dev/release/publish/actions/branch-off-next-branch.ts

```diff
@@ -44,30 +44,36 @@ export abstract class BranchOffNextBranchBaseAction extends ReleaseAction {
   }
 
   override async perform() {
+    const nextBranchName = this.active.next.branchName;
     const compareVersionForReleaseNotes = await getReleaseNotesCompareVersionForNext(
       this.active,
       this.config,
     );
     const newVersion = await this._computeNewVersion();
     const newBranch = `${newVersion.major}.${newVersion.minor}.x`;
+    const beforeStagingSha = await this.getLatestCommitOfBranch(nextBranchName);
```

Comment:
Shouldn't this value be created during the build method as its meant to validate what was used to actually build?

Particularly in this case as a whole new branch is being created.

Reply:
I thought a lot about this, but we would basically be making multiple requests for the same information (getting the latest SHA of the upstream branch). Also considered just using the local `rev-parse HEAD` once we checked out the remote branch (to save the second API call -- but this is inconsistent and has other downsides -- happy to expand).

All actions require awareness of how the staging / publishing works anyway, and this branch-off action heavily relies on that information already, so to save some additional requests, and to avoid having to mock a duplicate second GitHub request, I decided to just call it `beforeStagingSha` and require it to be passed manually when staging is performed in an advanced way (like done here in the banch-off scenario)

Reply:
Okay, sounds good